src/config_magic.c

```diff
@@ -1784,7 +1785,33 @@ TbBool parse_magic_shot_blocks(char *buf, long len, const char *config_textname,
                   COMMAND_TEXT(cmd_num), blocknamelen, blockname, config_textname);
           }
           break;
-      case ccr_comment:
+        case 61: // SPREAD_CIRCULAR (Combined scattering)
+            if (get_conf_parameter_single(buf, &pos, len, word_buf, sizeof(word_buf)) > 0) {
+                k = atoi(word_buf);
+                shotst->spread_circular = k; // Set the value for combined scattering
+                n++;
+
+                // Generate random position within a circle for XY
+                float angle = ((float)rand() / RAND_MAX) * 2 * M_PI; // Random angle between 0 and 2π
```

Comment:
We do not use floats in any code that should remain in sync, and projectiles are certainly among that. No rand() either, use our synced random functions. This will break packetsave and multiplayer.